src/main/java/bio/terra/service/filedata/flight/ingest/IngestFileFileStep.java

```diff
@@ -32,32 +32,35 @@ public IngestFileFileStep(FireStoreDao fileDao,
 
     @Override
     public StepResult doStep(FlightContext context) {
-        FlightMap inputParameters = context.getInputParameters();
-        FileLoadModel fileLoadModel = inputParameters.get(JobMapKeys.REQUEST.getKeyName(), FileLoadModel.class);
-
         FlightMap workingMap = context.getWorkingMap();
-        FSFileInfo fsFileInfo = workingMap.get(FileMapKeys.FILE_INFO, FSFileInfo.class);
-        String fileId = workingMap.get(FileMapKeys.FILE_ID, String.class);
+        Boolean loadComplete = workingMap.get(FileMapKeys.LOAD_COMPLETED, Boolean.class);
+        if (loadComplete == null || !loadComplete) {
```

Comment:
This wasn't an issue before because this is adding support for re-doing a bulk load when it has previously failed. You can read more about it in the design doc here: https://docs.google.com/document/d/1RPYuQ8gp8ipO1Vy4RfX84vbdyg_wIFO6syckuurTZbU/edit?folder=1njMO20WaretNaZ7NxnKGyu20N9VmhB6K&pli=1#

The gist is, we mark the directory entry and file entry with the loadTag. If the same loadTag is used to load the same file, and the file already exists, we skip the loading work and claim success. That code fragment is doing the skipping.